### PR TITLE
SocketWrapper Memory Leak

### PR DESCRIPTION
**Pull request by datasage**
As far as I can tell, this array was only being used to provide a connection count. To save performance on adding/removing sockets from the array, I turned this into a counter.

This should also fix a memory leak.

### CODE REVIEW

**Comment by ronag** [comment]
Nice find!

File: src/message/connection-endpoint.js [hunk 1]
```diff
@@ -369,7 +369,7 @@ module.exports = class ConnectionEndpoint extends events.EventEmitter {
       this.emit('client-connected', socketWrapper)
     }
 
-    this._authenticatedSockets.push(socketWrapper)
+    this._authenticatedSocketsCounter++;
```
Comment (ronag) [inline src/message/connection-endpoint.js]: no semi colon

File: src/message/connection-endpoint.js [hunk 2]
```diff
@@ -54,7 +54,7 @@ module.exports = class ConnectionEndpoint extends events.EventEmitter {
     this._ws.on('error', this._onError.bind(this))
     this._ws.on('connection', this._onConnection.bind(this))
 
-    this._authenticatedSockets = []
+    this._authenticatedSocketsCounter = 0
```
Comment (ronag) [inline src/message/connection-endpoint.js]: Maybe rename to `_connectionCount`?

Comment (yasserf) [inline src/message/connection-endpoint.js]: Given this gets incremented during the authentication callback makes sense to have it as `authenticatedSocketsCounter`

**Comment by yasserf** [comment]
Thanks!

**Pull request merged**